apps/s_cb.c

```diff
@@ -1355,3 +1356,41 @@ void ssl_ctx_security_debug(SSL_CTX *ctx, int verbose)
     SSL_CTX_set_security_callback(ctx, security_callback_debug);
     SSL_CTX_set0_security_ex_data(ctx, &sdb);
 }
+
+static void keylog_callback(const SSL *ssl, const char *line)
+{
+    /*
+     * There might be concurrent writers to the keylog file, so we must ensure
+     * that one line is written at once.
+     */
+    BIO_puts(bio_keylog, line);
+    BIO_write(bio_keylog, "\n", 1);
+    (void)BIO_flush(bio_keylog);
+}
+
+int set_keylog_file(SSL_CTX *ctx, const char *keylog_file)
+{
+    /* Close any open files */
+    BIO_vfree(bio_keylog);
+    bio_keylog = NULL;
+
+    if (ctx == NULL || keylog_file == NULL) {
+        /* Keylogging is disabled, OK. */
+        return 0;
+    }
+
+    bio_keylog = BIO_new_file(keylog_file, "a");
```

Comment:
Yes, in order to preserve existing keylog files (mirrors SSLKEYLOGFILE behavior). You can then run `openssl s_client` multiple times and the key log file will keep growing.